### PR TITLE
[CSL-1770] Integrate report server 0.4

### PR DESCRIPTION
**Pull request by volhovM**
I've also decided to put git revision into `rBuild` report field so it required me to pass `HasCompileInfo` in several places. Tests and block-gen use fake compile info (default instance) as they never report anything (though using methods that can report).

### CODE REVIEW

File: lib/src/Pos/Generator/Block/Logic.hs [hunk 1]
```diff
@@ -117,10 +118,12 @@ genBlock eos = do
                     Just <$> usingPrimaryKey leaderSK (genMainBlock slot (Right . swap <$> transCert))
   where
     genMainBlock slot proxySkInfo =
+        withCompileInfo def $
         lift $ createMainBlockInternal @SscGodTossing slot proxySkInfo >>= \case
             Left err -> throwM (BGFailedToCreate err)
             Right mainBlock -> verifyAndApply $ Right mainBlock
     verifyAndApply block =
+        withCompileInfo def $
```
Comment (gromakovsky) [inline lib/src/Pos/Generator/Block/Logic.hs]: Looks strange to have 3 `withCompileInfo` in `genBlock`. Maybe you need to provide type signatures explicitly.

Comment (volhovM) [reply lib/src/Pos/Generator/Block/Logic.hs]: I think this difference is mostly cosmetical, but i can do it, sure.

Comment (gromakovsky) [reply lib/src/Pos/Generator/Block/Logic.hs]: The problem is that if `withCompileInfo` is applied to the same code multiple times with different values, something bad will happen, AFAIK (and it won't be prevented by compiler).
So having less `withCompileInfo` makes reasoning easier.
Later someone may change `def` to something else in one place and forget about another one.

Comment (volhovM) [reply lib/src/Pos/Generator/Block/Logic.hs]: In general -- yes. This particular place -- no. But anyway, I agree.

**Pull request merged**